### PR TITLE
Add Logout Account

### LINKED ISSUE #1296: New user will see notes of the previous user.
### Expected
New user should see only the notes belonging to them.

### Observed
After sign up or logging in, the user will see the notes of the previously logged-in user. This is 'only' a display thing - removing these notes will not affect the real notes of the owner. Also, logging in under same account on a different device (iOS) will not show them.

### Reproduced
See video.

1. Be logged in as a verified user with some amount of notes
2. Log out
3. Sign up / log in ~> you will see the notes of previous user 

https://user-images.githubusercontent.com/73365754/106445585-de6d4e00-6487-11eb-8c22-42ff0158393e.mp4

Device Make: Samsung
Device Model: Galaxy A31
Android Version: 10
Simplenote Version: 2.16-rc-2

### PR DESCRIPTION
**Pull request by theck13**
### Fix
Add resetting and stopping the `Account` bucket when logging out to mimic the other buckets and close #1296.  Even though the `Account` bucket does not contain note data, that data is being kept if the `Account` bucket is not reset and stopped.

### Test
Since the issue involves showing the previous account's notes after logout/login, two different Simplenote accounts with different note data is required to confirm the fix.  They are labeled Account 1 and Account 2 in the steps below.
1. Notice notes from ***Account 1***.
2. Open navigation drawer.
3. Tap ***Settings*** in navigation drawer.
4. Tap ***Log out*** under ***Account*** section.
5. Tap ***Log In*** button.
6. Tap ***Log In with Email*** button.
7. Enter email and password in ***Email*** and ***Password*** fields for ***Account 2***.
8. Tap ***Log In*** button.
9. Notice note from ***Account 2***.

### Review
Only one developer is required to review these changes, but anyone can perform the review.

### Release
These changes do not require release notes.

#### Note
@mokagio, these changes will require a new release candidate build once they are merged.

### CODE REVIEW

**Review by mokagio — APPROVED** [review]
I reproduced the issue on the beta downloaded from the Store on my device.

I reproduced the fix from this branch on the Emulator. I tried three different accounts and also verified that changes to the notes sync after logging out and in to a different account.

The code change is minimal so I feel entitle to leave a ✅ . I'll leave it up to you whether to merge it or wait for other Android reviewers to chime in.

Regardless, I receive notifications for this PR, so I'll push a new RC as soon as it lands.

**Comment by theck13** [comment]
@mokagio, thanks for the approval!  I think this is good to merge whenever you're ready.

**Comment by pachlava** [comment]
[testing] fix verified on Samsung Galaxy A31 | Android 10 | Simplenote 2.16-rc-3

**Pull request merged**
